pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -88,6 +88,10 @@ type Manager interface {
 
 	// GetAllocatableCPUs returns the assignable (not allocated) CPUs
 	GetAllocatableCPUs() cpuset.CPUSet
+
+	// GetCPUsetOrDefault returns cpuset which includes cpus from shared pools
+	// as well as exclusively allocated cpus
+	GetCPUsetOrDefault(podUID, containerName string) cpuset.CPUSet
```

Comment:
Assuming this is true, I feel like better names for these API calls would be:

1.  `GetExclusiveCPUs()`
2.  `GetCPUAffinity()` 

Where `GetExclusiveCPUs()` returns the list of CPUs exclusive to this container, and `GetCPUAffinity()` returns the full list of CPUs this container is able to run on (i.e. the CPUs it is affinitized to). If `GetExclusiveCPUs() == GetCPUAffinity()`, then you know all CPUs assigned to the container are exclusive, otherwise they are not.

However, since we already have `GetCPUs()`as an API call, I am fine repurposing it to mean `GetExclusiveCPUs()` (instead of introducing a whole new call for this purpose). We should definitely update the comment to make this very clear though.

That said, I do think we should rename `GetCPUsetOrDefault()` to `GetCPUAffinity()`. This is much clearer in my opinion, and not as tied to the current implementation / naming of the call in the `state` abstraction.

I'm curious if / how this naming could be extended to devices as well. At present we don't allow sharing of devices, but nothing is preventing us from allowing this in the future. I guess `GetDevices()` could continue to return "exclusive" devices, but a call like `GetDeviceAffinity()` doesn't really make much sense (especially since this call returns a list of many different device types across different vendors). What would be a good name in this circumstance to mean "Shared+Exclusive"?

Reply:
> Just to make sure I'm understanding this correctly -- this new call returns what `GetCPUs()` used to return (i.e. the list of CPUs from `m.state.GetCPUSetOrDefault()`).
> 
> And the `GetCPUs()` call has now been changed to only return the list of _exclusive_ CPUs (if there are any).
> 
> Is this correct?

Yes, it's correct!

Reply:
> Assuming this is true, I feel like better names for these API calls would be:
> 
> 1. `GetExclusiveCPUs()`
> 2. `GetCPUAffinity()`
> 
> Where `GetExclusiveCPUs()` returns the list of CPUs exclusive to this container, and `GetCPUAffinity()` returns the full list of CPUs this container is able to run on (i.e. the CPUs it is affinitized to). If `GetExclusiveCPUs() == GetCPUAffinity()`, then you know all CPUs assigned to the container are exclusive, otherwise they are not.
> 
> However, since we already have `GetCPUs()`as an API call, I am fine repurposing it to mean `GetExclusiveCPUs()` (instead of introducing a whole new call for this purpose). We should definitely update the comment to make this very clear though.
> That said, I do think we should rename `GetCPUsetOrDefault()` to `GetCPUAffinity()`. This is much clearer in my opinion, and not as tied to the current implementation / naming of the call in the `state` abstraction.

Sure! I will update the names of the API calls as per your suggestion.

Reply:
> I'm curious if / how this naming could be extended to devices as well. At present we don't allow sharing of devices, but nothing is preventing us from allowing this in the future. I guess GetDevices() could continue to return "exclusive" devices, but a call like GetDeviceAffinity() doesn't really make much sense (especially since this call returns a list of many different device types across different vendors). What would be a good name in this circumstance to mean "Shared+Exclusive"?

This is interesting and we can prepare ourselves for a case like this in case of devices as well. How about we change the name of GetDevice() to GetExclusiveDevice() and then repurpose GetDevice() API call to correspond to "Shared+Exclusive" devices ?